### PR TITLE
Creation & implementation of Filter Sidebar

### PR DESCRIPTION
**Pull request by Denperidge**
- [x] Allow queryParam getting & setting without extra route/controller code needed
- [x] Update into agenda-items
- [x] Implement into sessions

### CODE REVIEW

File: app/controllers/agenda-items.ts [hunk 1]
```diff
@@ -64,51 +58,6 @@ export default class AgendaItemsController extends Controller {
...
-    this.router.transitionTo("agenda-items", {
-      queryParams: {
-        begin: start,
-        eind: end,
-      },
-    });
-  }
-
-  @action hideDatePicker(picker: any, start: any, end: any) {}
+  @action handleSort(e: any) {}
```
Comment (aliokan) [inline app/controllers/agenda-items.ts:61]: **issue: ** This line was removed on Friday https://github.com/lblod/frontend-burgernabije-besluitendatabank/pull/23/files

File: app/controllers/sessions/index.ts [hunk 2]
```diff
@@ -5,20 +5,33 @@ import { ModelFrom } from '../../lib/type-utils';
...
         //todo add max page guard
         if (this.model && !this.isLoadingMore) {
             this.isLoadingMore = true;
             const nextPage = this.model.currentPage + 1;
 
-            const sessions = await this.store.query("session", this.model.getQuery(nextPage));
+            let plannedStartMin = String(get(this, "plannedStartMin")) || undefined;
+            let plannedStartMax = String(get(this, "plannedStartMax")) || undefined;
+            let municipalities = String(get(this, "municipality")) || undefined;
+            console.log(plannedStartMax, plannedStartMax, municipalities)
```
Comment (aliokan) [inline app/controllers/sessions/index.ts:32]: here too

File: app/controllers/sessions/index.ts [hunk 3]
```diff
@@ -5,20 +5,33 @@ import { ModelFrom } from '../../lib/type-utils';
...
+
+            const sessions = await this.store.query("session", this.model.getQuery(nextPage, plannedStartMin, plannedStartMax, municipalities));
```
Comment (aliokan) [inline app/controllers/sessions/index.ts:34]: **suggestion:** It's a good practice to use named arguments when there is more than 2 arguments. https://dev.to/producthackers/named-arguments-js-2d99

File: app/routes/agenda-items.ts [hunk 4]
```diff
@@ -146,11 +146,13 @@ export default class AgendaItemsRoute extends Route {
     this.plannedStartMin = params.plannedStartMin || "";
     this.plannedStartMax = params.plannedStartMax || "";
 
+    /*
```
Comment (aliokan) [inline app/routes/agenda-items.ts:149]: **suggestion:** I guess these lines can be removed

File: app/routes/searcher.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import Store from "@ember-data/store";
```
Comment (aliokan) [inline app/routes/searcher.ts:1]: **question:** Is this file used?

**Review by aliokan — CHANGES_REQUESTED** [review]
Nice work :+1: 

I left some comments, only the **issue** should be considered blocking, the other comments are not blocking

File: app/components/filter-sidebar/date-range-filter.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import { tracked } from '@glimmer/tracking';
+import FilterComponent from './filter';
+
+export default class SearcherDateRangeFilterComponent extends FilterComponent {
+
+
+
+  @action
+  async dateChange(e: any, start: string, end: string) {
+    console.log(start)
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/date-range-filter.ts:12]: We do not!

File: app/components/filter-sidebar/filter.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+export default class FilterComponent extends Component<ArgsInterface> {
+    @service declare router: RouterService;
+
+    updateQueryParams(params: object) {
+        this.router.transitionTo(this.router.currentRouteName, {
+            queryParams: params,
+        });
+    }
+
+    /*
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/filter.ts:25]: Done!

File: app/components/filter-sidebar/date-range-filter.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import Component from '@glimmer/component';
+import { action } from "@ember/object";
+import { tracked } from '@glimmer/tracking';
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/date-range-filter.ts:3]: Done!

File: app/components/filter-sidebar/select-filter.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { action } from "@ember/object";
+import FilterComponent from './filter';
+import { get } from '@ember/object';
+
+export default class SearcherSelectFilterComponent extends FilterComponent {
+    @action
+    async selectChange(m: any) {
+        let value = m ? get(m, this.args.searchField) : ""
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-filter.ts:8]: You are correct! Null seems to do the trick. I remember it not working, but that could've been caused by something else at the time

File: app/controllers/agenda-items.ts (hunk 1 above)
Comment (Denperidge) [reply app/controllers/agenda-items.ts:61]: Oh woops, I'll remove it from there too!

File: app/controllers/sessions/index.ts [hunk 10]
```diff
@@ -5,20 +5,33 @@ import { ModelFrom } from '../../lib/type-utils';
...
+    @service declare municipalityList: MunicipalityListService;
+
+    get municipalities() {
+        return this.municipalityList.municipalities();
+    }
 
     @action
     async loadMore() {
+        console.log(get(this.model, "municipality"));
+        console.log(get(this, "municipality"));
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/controllers/sessions/index.ts:23]: Done!

File: app/controllers/sessions/index.ts (hunk 2 above)
Comment (Denperidge) [reply app/controllers/sessions/index.ts:32]: Done pt2!

File: app/controllers/sessions/index.ts (hunk 3 above)
Comment (Denperidge) [reply app/controllers/sessions/index.ts:34]: I tried, but TypeScript is being a bit difficult about it. Might revisit it later, or let it be as is for now

File: app/routes/agenda-items.ts (hunk 4 above)
Comment (Denperidge) [reply app/routes/agenda-items.ts:149]: Done!

File: app/routes/searcher.ts (hunk 5 above)
Comment (Denperidge) [reply app/routes/searcher.ts:1]: It is not! A leftover I missed, my apologies

File: app/components/filter-sidebar/select-filter.hbs [hunk 11]
```diff
@@ -0,0 +1,26 @@
+<FilterSidebar::Filter @id={{@id}} @label={{@label}}>
+    <PowerSelect
+      style="width: 100%;"
```
[the remark this answers is not in the document]
Comment (Denperidge) [reply app/components/filter-sidebar/select-filter.hbs]: Implemented!

File: app/components/filter-sidebar/text-filter.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+
+export default class TextFilterComponent extends FilterComponent {
+    
+    @action
+    async onChange(e: Event) {
+        if (e.target) {
+            let value = (e.target as HTMLInputElement).value;
+            console.log("start")
+            this.updateQueryParams({
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/components/filter-sidebar/text-filter.ts:10]: follow-up: https://binnenland.atlassian.net/browse/BNB-210

**Comment by aliokan** [comment]
@Denperidge In order to move forward, I am merging this PR. Regarding the debounce, I created a follow-up ticket: https://binnenland.atlassian.net/browse/BNB-210

**Pull request merged**